internal/session/session.go

```diff
@@ -139,6 +139,13 @@ func (s *Session) serve(ctx context.Context) error {
 
 	for {
 		select {
+		case stateUpdate := <-s.state.GetStateUpdatesCh():
+			if err := s.state.ApplyUpdate(ctx, stateUpdate); err != nil {
+				logrus.WithError(err).Error("Failed to apply state update")
+			}
+
+			continue
+
```

Comment:
Any reason to move this above? just checking that you don't expect it to change behaviour

Reply:
Yes. If we move this up we can guarantee that the update will be processed before the next command comes in. I had some flaky tests because some updates were processed in between commands. This sort of fixes the issue.

Reply:
As discussed, order of select branches doesn't matter. But moving it here can't harm anything 🤷